### PR TITLE
Require that commands are passed in as static strings

### PR DESCRIPTION
**Pull request by andydotxyz**
This is a technique I learned at GoLab and it seems like a great way to avoid command injection

### CODE REVIEW

**Comment by Jacalz** [comment]
Wow. This is a very cool idea 💡:)

File: shell.go [hunk 1]
```diff
@@ -7,14 +7,17 @@ import (
 	"strings"
 )
 
-// CommandInShell sets up a new command with the environment set up by users environment.
+type staticString string
```
Comment (Jacalz) [inline shell.go:10]: I think the right term is "string literal" in this case. Maybe even a comment for the type would be useful if anyone is confused?

Comment (andydotxyz) [reply shell.go:10]: Thanks

**Review by Bluebugs — APPROVED** [review]
Nice trick.

File: shell.go [hunk 2]
```diff
@@ -7,14 +7,19 @@ import (
...
+// stringLiteral is a type of string that cannot be passed as a variable.
+// This helps to avoid code injection by enforcing executable names to be hard coded.
+type stringLiteral string
+
+// CommandInShell sets up a new command with the environment set up by user's environment.
 // In darwin or other Unix systems the shell will be loaded by running the terminal and accessing env command.
-// The command is run directly so stdout and stderr can be read directly from the returnd `exec.Cmd`.
-func CommandInShell(cmd string, args ...string) *exec.Cmd {
+// The command is run directly so stdout and stderr can be read directly from the returned `exec.Cmd`.
+// The cmd parameter must be a string constant to avoid possible code injection attacks.
```
Comment (Jacalz) [inline shell.go:17]: Does a string constant work or is it just string literals? If it's the latter, we should probably say literal instead.

Comment (andydotxyz) [reply shell.go:17]: Constants and literals both work

**Review by Jacalz — APPROVED** [review]

**Pull request merged**